src/Streamly/Internal/Data/Stream/Zip.hs

```diff
@@ -96,9 +99,10 @@ zipWithK f = zipWithMK (\a b -> return (f a b))
 {-# INLINE zipAsyncWithMK #-}
 zipAsyncWithMK :: MonadAsync m
     => (a -> b -> m c) -> Stream m a -> Stream m b -> Stream m c
-zipAsyncWithMK f m1 m2 = D.toStreamK $
-    D.zipWithM f (Par.mkParallelD $ D.fromStreamK m1)
-                 (Par.mkParallelD $ D.fromStreamK m2)
```

Comment:
This seems cleaner! We can achieve this using unfold and the composition of unfolds?

Reply:
Composition of `inject` would do the trick of initializing the parallel svars.